pkg/apis/sources/v1alpha1/vspheresource_defaults.go

```diff
@@ -32,4 +32,8 @@ func (vs *VSphereSource) SetDefaults(ctx context.Context) {
 	} else {
 		vs.Spec.PayloadEncoding = strings.ToLower(vs.Spec.PayloadEncoding)
 	}
+
+	if vs != nil && vs.Spec.ServiceAccountName == "" {
+		vs.Spec.ServiceAccountName = "default"
```

Comment:
We don't need to set the serviceAccountName. A pod created without a service account would just use the default service account for that namespace. It might be cleaner to just leave it optional, don't default it and set a `""` on the deployment's pod spec

Reply:
I see this practical for when creating the rolbinding

Reply:
I don't think I follow. What does it have to do with rolebindings?

Reply:
https://github.com/vmware-tanzu/sources-for-knative/blob/26d67c594c687208db8c7e87612df101592bd859/pkg/reconciler/vspheresource/resources/role_binding.go#L18

Reply:
oh right. Thanks

Reply:
I think in that case it might be better to leave the naming function there with the same logic. Sorry for the back n forth but that method would be more explicit than a webhook defaulting. If in the future someone deems this defaulting pointless, it would break the rolebinding that relies on it.

Reply:
Kk, and no problem at all with the back and forth :p, so just to see if I follow:
- remove the defaulting
- Keep the naming function
- Still set the empty serviceaccount name to default

Reply:
Yep sounds good 👍